app/forms/bulk_update/bulk_add_trainees_upload_form.rb

```diff
@@ -22,14 +22,18 @@ def initialize(provider: nil, file: nil)
     def save
       return false unless valid?
 
-      BulkUpdate::TraineeUpload.create(
+      @upload = BulkUpdate::TraineeUpload.create(
         provider: provider,
-        file: file,
+        file: File.read(file),
         file_name: file.original_filename,
         number_of_trainees: csv&.count,
         status: valid? ? :pending : :failed,
         error_messages: errors.messages.values.inject([], &:concat),
       )
+
+      BulkUpdate::AddTrainees::ImportRowsJob.perform_later(id: @upload.id)
```

Comment:
The upload object could be passed to the job as an object, it will be serialised and be found by Global ID inside the worker. More info could be found at https://github.com/rails/globalid/blob/main/README.md

```suggestion
      BulkUpdate::AddTrainees::ImportRowsJob.perform_later(upload)
```